sklearn/compose/tests/test_column_transformer.py

```diff
@@ -1266,3 +1266,19 @@ def test_make_column_selector_pickle():
     selector_picked = pickle.loads(pickle.dumps(selector))
 
     assert_array_equal(selector(X_df), selector_picked(X_df))
+
+
+def test_feature_names_empty_columns():
+    pd = pytest.importorskip('pandas')
+
+    df = pd.DataFrame({"col1": ["a", "a", "b"], "col2": ["z", "z", "z"]})
+
+    ct = ColumnTransformer(
+        transformers=[
+            ("ohe", OneHotEncoder(), ["col1", "col2"]),
+            ("empty_features", OneHotEncoder(), []),
```

Comment:
I think you should also test with a callable that returns empty rather than relying on an implementation detail